integration-tests/src/tests/client/process_blocks.rs

```diff
@@ -2804,6 +2784,120 @@ fn test_epoch_protocol_version_change() {
     assert_eq!(protocol_version, PROTOCOL_VERSION);
 }
 
+#[test]
+fn test_epoch_multi_protocol_version_change() {
+    init_test_logger();
+
+    let v0 = PROTOCOL_VERSION - 2;
+    let v1 = PROTOCOL_VERSION - 1;
+    let v2 = PROTOCOL_VERSION;
+
+    // produce blocks roughly every 100ms
+    // one epoch is roughly 500ms
```

Comment:
such time-based tests always make me nervous that something is going to go wrong and annoy people with unnecessary failures in the future, but i guess there's probably enough leeway that i doubt it will happen

Reply:
Yeah I feel the same but nonetheless it seemed the most practical.